apps/prairielearn/python/zygote.py

```diff
@@ -46,21 +46,33 @@
 #
 # To work around this, we'll set `$XDG_CONFIG_HOME` and `$XDG_CACHE_HOME` to
 # directories created in `/tmp` that are world-writable. matplotlib and
-# fontconfig should respect these environment variables; other tools should as
-# well. If they don't, special cases can be added below.
+# fontconfig should respect these environment variables; other tools should too.
+#
+# Note that `mktexfmt` does _not_ respect those environment variables, so we'll
+# also set `$TEXMFCONFIG`, `$TEXMFVAR`, and `$TEXMFHOME` to directories created
+# in `/tmp` that are world-writable. This allows LaTeX to run properly.
```

Comment:
It occurs to me that we could just set `$HOME` to some directory that _is_ writable by the `executor` user (e.g. `/home/executor`). That might be a more generic fix?

Reply:
Isn't this comment stating that fix wouldn't be generic because latex ignores the above environment variables?

Reply:
LaTeX respects `$HOME`, which is currently `/root`, and assumes that that directory is writable. I'm proposing we should make that something else.

Reply:
Ahhh I see. Maybe that's the way to go? I guess there aren't any expectations on what's stored there by client code so something like that is probably fine.

Reply:
Would `/home/executor` be shared between executions?

Reply:
Yes, though `/tmp` is as well.

Reply:
I like the idea of setting `$HOME` to a writeable directory and then explicitly deleting everything in it between executions.